### ISSUE TITLE
[RangeSlider] Add dual input

### ISSUE THREAD

**Issue by solonaarmstrong** [Shopify/polaris-react#595]
# Feature request summary

Add a second input (button) to the existing range slider component to create a range of values instead of only one value.

![screen shot 2018-11-09 at 3 20 57 pm](https://user-images.githubusercontent.com/6239480/48293281-2b8e2100-e433-11e8-90be-5f28d297b456.jpg)

To make this accessible, the two input values will each need to be accompanied by text (number) inputs that can control the values.

## Rationale

This feature is needed for some list filters. For example:
- To select customers with a range of money spent
- To select customers with a range of number of orders placed

**Comment by AndrewMusgrave** [comment]
Rather than an addition prop, have we considered value accepting number or tuple? For example
```tsx
interface Props {
  ...
  value: number | [number, number]
  ...
}
```

**Comment by solonaarmstrong** [comment]
@AndrewMusgrave When I was exploring this, I created a `Range` interface, similar to `DatePicker`, but I think your suggestion is better. 👍

**Comment by solonaarmstrong** [comment]
@tmlayton This is a good starting point for a technical design.

**Comment by dpersing** [comment]
A few notes for a11y that we've discussed elsewhere but thought I'd include here for technical design discussion:

- There is an [ARIA 1.1 pattern for a multi-thumb range slider](https://www.w3.org/TR/wai-aria-practices/#slidertwothumb), but there's no equivalent in HTML, so it relies entirely on ARIA and JavaScript for semantics and functionality. This should be fine for keyboard users, but might get hairy for desktop screen reader users.
- ARIA-only range sliders, including single-thumb ones, aren't supported well by mobile screen readers. VoiceOver and TalkBack really only seem to like sliders that build off the native `input="range"` and choke on spec-perfect single-thumb ARIA-only sliders.
- There's no native touch gesture for adjusting a multi-thumb slider, since the screen reader cursor focus goes to the whole slider and the user swipes up and down to move the handle/thumb. The ARIA multi-thumb pattern doesn't currently include any touch/mobile info.

A good progressive enhancement solution for setting any range value is a labeled text input or set of inputs for a multi-thumb slider. Letting users enter values in the text inputs or via the slider and having the values update either way for both sets of controls would be ideal.

**Comment by solonaarmstrong** [comment]
CC @tmlayton
